src/github.com/matrix-org/dendrite/syncserver/storage/output_room_events_table.go

```diff
@@ -232,7 +248,14 @@ func (s *outputRoomEventsStatements) Events(txn *sql.Tx, eventIDs []string) ([]g
 	return result, nil
 }
 
-func rowsToEvents(rows *sql.Rows) ([]gomatrixserverlib.Event, error) {
+type rowOrdering uint8
+
+const (
+	rowOrderForward = rowOrdering(iota)
+	rowOrderReverse
+)
```

Comment:
Hmm, I see. you want to apply an order by with a limit to get the last N, and then reverse the order. that kind of makes sense. I'd suggest writing a reverse function that takes a list of events and reverses the order of the list. Rather than adding the enum to this function.